File: ios/SplitBackAPI/Sources/SplitBackAPI/Models/Enums.swift
import Foundation

/// Backend that owns a group. Raw values match the API (`self_hosted` | `splitwise`).
enum BackendType: String, Codable, CaseIterable, Sendable {
    case selfHosted = "self_hosted"
    case splitwise = "splitwise"
}

/// Origin of a transaction. Raw values match the API (`plaid` | `manual`).
enum TransactionSource: String, Codable, CaseIterable, Sendable {
    case plaid = "plaid"
    case manual = "manual"
}

/// Where a user/person record came from. Raw values match the API.
/// `app` = a household member with an API token; `manual` = added in-app; `splitwise` = imported.
enum UserSource: String, Codable, CaseIterable, Sendable {
    case app = "app"
    case manual = "manual"
    case splitwise = "splitwise"
}

File: backend/app/models/enums.py
import enum


class BackendType(str, enum.Enum):
    self_hosted = "self_hosted"
    splitwise = "splitwise"


class TransactionSource(str, enum.Enum):
    plaid = "plaid"
    manual = "manual"


class UserSource(str, enum.Enum):
    app = "app"
    manual = "manual"
    splitwise = "splitwise"


class NotificationSource(str, enum.Enum):
    splitwise = "splitwise"
    app = "app"


class ConnectionStatus(str, enum.Enum):
    pending = "pending"
    accepted = "accepted"


class ShareLevel(str, enum.Enum):
    private = "private"      # owner only
    balances = "balances"    # partner sees the balance, not transactions
    full = "full"            # partner sees balance + transactions
